server/resp/src/main/java/org/infinispan/server/resp/scripting/LuaContext.java

```diff
@@ -0,0 +1,345 @@
+         } else if (!l.isNumber(-argc)) {
+            luaPushError(lua, "First argument must be a number (log level).");
+            return -1;
+         }
+         int level = (int) l.toInteger(-argc);
+         if (level < LOG_DEBUG || level > LOG_WARNING) {
+            luaPushError(lua, "Invalid log level.");
+            return -1;
+         }
+         StringBuilder sb = new StringBuilder();
+         for (j = 1; j < argc; j++) {
+            sb.append(l.toString(j - argc));
+         }
+         Log.SERVER.log(LEVEL_MAP[level], sb);
+         return 0;
+      });
+      tableAdd(lua, "LOG_DEBUG", LOG_DEBUG);
+      tableAdd(lua, "LOG_VERBOSE", LOG_VERBOSE);
+      tableAdd(lua, "LOG_NOTICE", LOG_NOTICE);
+      tableAdd(lua, "LOG_WARNING", LOG_WARNING);
+      tableAdd(lua, "REDIS_VERSION_NUM", Version.getVersionShort());
+      tableAdd(lua, "REDIS_VERSION", Version.getVersion());
+      // Give a name to the table
+      lua.setGlobal(REDIS_API_NAME);
+   }
+
+   private void installErrorHandler() {
+      String err_handler = """
+            local dbg = debug
+            debug = nil
+            function __redis__err__handler(err)
+              local i = dbg.getinfo(2,'nSl')
+              if i and i.what == 'C' then
+                i = dbg.getinfo(3,'nSl')
+              end
+              if type(err) ~= 'table' then
+                err = {err='ERR ' .. tostring(err)}
+              end
+              if i then
+                err['source'] = i.source
+                err['line'] = i.currentline
+              end
+              return err
+            end
+            """;
+      byte[] bytes = err_handler.getBytes(StandardCharsets.US_ASCII);
+      ByteBuffer buffer = ByteBuffer.allocateDirect(bytes.length);
+      buffer.put(bytes);
+      lua.load(buffer, "@err_handler_def");
+      lua.pCall(0, 0);
+   }
+
+   public int executeRespCommand(Lua l, boolean raiseError) {
+      int argc = l.getTop();
+      String command = l.toString(-argc);
+      RespCommand respCommand = RespCommand.fromString(command);
+      if (respCommand == null) {
+         l.push("Unknown Redis command called from script");
+         return -1;
+      }
+      long commandMask = respCommand.aclMask();
+      if ((commandMask & AclCategory.CONNECTION) == AclCategory.CONNECTION) {
+         l.push("This Redis command is not allowed from script");
+         return -1;
+      }
+      if (ScriptFlags.NO_WRITES.isSet(flags) && (commandMask & AclCategory.WRITE) == AclCategory.WRITE) {
+         l.push("Write commands are not allowed from read-only scripts.");
+         return -1;
+      }
+      List<byte[]> args = new ArrayList<>(argc - 1);
+      for (int i = -argc + 1; i < 0; i++) {
+         args.add(l.toString(i).getBytes(StandardCharsets.US_ASCII));
+      }
+      CompletableFuture<RespRequestHandler> future = ((Resp3Command) respCommand).perform(handler, ctx, args).toCompletableFuture();
+      try {
+         future.get();
```

Comment:
This needs to be evaluated for blocking in some fashion.

Reply:
Simplest way would just be to use the BlockingManager.runBlocking surrounding the eval and evalSha method contents. Then those methods can return a CompletionStage<Void> which should be chained appropriately.

Reply:
https://github.com/wburns/infinispan/commit/a54b28753bd49cb5c14607f670fe4389becb2dc3

Something like this should work.